cpp/log/cms_verifier.cc

```diff
@@ -228,15 +228,15 @@ Cert* CmsVerifier::UnpackCmsSignedCertificate(const string& cms_object) {
   BIO_write(source_bio.get(), cms_object.c_str(), cms_object.length());
 
   ScopedBIO unpacked_bio(BIO_new(BIO_s_mem()));
-  unique_ptr<Cert> cert(new Cert);
+  unique_ptr<Cert> cert;
 
   if (UnpackCmsDerBio(source_bio.get(), unpacked_bio.get()).ok()) {
     // The unpacked data should be a valid DER certificate.
     // TODO: The RFC does not yet define this as the format so this may
     // need to change.
-    const Status status = cert->LoadFromDerBio(unpacked_bio.get());
+    cert = Cert::FromDerBio(unpacked_bio.get());
 
-    if (!status.ok()) {
+    if (!cert) {
       LOG(WARNING) << "Could not unpack cert from CMS DER encoded data";
```

Comment:
Yeah, with the previous version, it would have returned a non-null `Cert*`, which would have returned false from `IsLoaded()`? But would have returned null if `UnpackCmsDerBio` had an error? Seems a bit dodgy indeed!

The new version will at least return null in both error cases, and non-null only in case of success.